src/uct/ib/mlx5/dv/ib_mlx5dv_md.c

```diff
@@ -1580,8 +1688,98 @@ uct_ib_mlx5_devx_reg_exported_key(uct_ib_md_t *ib_md, uct_ib_mem_t *ib_memh)
     return uct_ib_mlx5_devx_xgvmi_umem_mr(md, memh);
 
 out:
-    memh->cross_mr            = cross_mr;
-    memh->super.exported_lkey = exported_lkey;
+    memh->cross_mr      = cross_mr;
+    memh->exported_lkey = exported_lkey;
+    return UCS_OK;
+}
+
+static ucs_status_t
+uct_ib_mlx5_devx_mkey_pack(uct_md_h uct_md, uct_mem_h uct_memh,
+                           const uct_md_mkey_pack_params_t *params,
+                           void *mkey_buffer)
+{
+    uct_ib_mlx5_md_t *md         = ucs_derived_of(uct_md, uct_ib_mlx5_md_t);
+    uct_ib_mlx5_devx_mem_t *memh = uct_memh;
+    ucs_status_t status;
+    unsigned flags;
+    uint32_t rkey;
+
+    flags = UCS_PARAM_VALUE(UCT_MD_MKEY_PACK_FIELD, params, flags, FLAGS, 0);
+    if (flags & UCT_MD_MKEY_PACK_FLAG_EXPORT) {
+        if (ENABLE_PARAMS_CHECK &&
```

Comment:
imo we could avoid checking `ENABLE_PARAMS_CHECK` inside this func, also it is under EXPORT check

Reply:
IMO EXPORT is still fast path because we could be exporting key from rcache